source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -129,6 +129,22 @@ Config::Config(const envoy::extensions::filters::network::tcp_proxy::v3::TcpProx
   if (!config.hash_policy().empty()) {
     hash_policy_ = std::make_unique<Network::HashPolicyImpl>(config.hash_policy());
   }
+
+  if (config.has_on_demand() && config.on_demand().has_odcds_config()) {
+    odcds_ = context.clusterManager().allocateOdCdsApi(config.on_demand().odcds_config(),
+                                                       OptRef<xds::core::v3::ResourceLocator>(),
+                                                       context.messageValidationVisitor());
+    if (config.on_demand().has_timeout()) {
```

Comment:
This entire chunk should be replaced with `PROTOBUF_GET_MS_OR_DEFAULT`, and a proto-validation to say that the timeout value must be greater than zero.

Reply:
using PROTOBUF_GET_MS_OR_DEFAULT now. The api validation will be added in the PR that defining and implementing per tcp proxy instance timeout